src/Raven.Server/RavenServer.cs

```diff
@@ -3009,7 +3009,7 @@ internal static void VerifyLicense(StorageEnvironment storageEnvironment, Server
                         var localLicenseStatus = LicenseManager.GetLicenseStatus(localLicense);
                         if (localLicenseStatus.Expiration >= RavenVersionAttribute.Instance.ReleaseDate)
                         {
-                            serverStore.LicenseManager.OnBeforeInitialize += () => serverStore.LicenseManager.TryActivateLicenseAsync(throwOnActivationFailure: serverStore.Server.ThrowOnLicenseActivationFailure).Wait(serverStore.ServerShutdown);
+                            serverStore.LicenseManager.OnBeforeInitialize += () => AsyncHelpers.RunSync(() => serverStore.LicenseManager.TryActivateLicenseAsync(throwOnActivationFailure: serverStore.Server.ThrowOnLicenseActivationFailure));
```

Comment:
We no longer pass `serverStore.ServerShutdown`

Reply:
Also on 6.0 we have one more `TryActivateLicenseAsync().Wait()` call:

https://github.com/ravendb/ravendb/blob/8ce4c5ea0d89ab3bee58f11d57acb40ca3921760/src/Raven.Server/RavenServer.cs#L3063-L3069

Reply:
yes, if we cancel we should throw from within the `TryActivateLicenseAsync`
and in 6.0 we will need have the this on both places

Reply:
Just to make sure - is it really the case that `TryActivateLicenseAsync` will throw then?

Reply:
added cancellation tokens